source/Mission.cpp

```diff
@@ -123,6 +123,20 @@ void Mission::Load(const DataNode &node)
 				passengerLimit = child.Value(2);
 			if(child.Size() >= 4)
 				passengerProb = child.Value(3);
+			for(const DataNode &grand : child)
```

Comment:
I would bundle this with the above, by defining a private method:
```
    ...
    if(child.Size() >= 4)
        passengerProb = child.Value(3);
    if(child.HasChildren())
        MarkIllegal(child);
}
...
void MarkIllegal(const DataNode &child)
{
    for(const DataNode &grand : child)
    {
        if(grand.Token(0) == "illegal" ...
    ...
```
You'd then call this method for `cargo` as well, removing code duplication for illegal stuff, since the same values are set for cargo and passengers in your implementation.